Sources/Datadog/Core/Persistence/Files/File.swift

```diff
@@ -43,23 +43,27 @@ internal struct File: WritableFile, ReadableFile {
     }
 
     /// Appends given data at the end of this file.
-    func append(transaction: ((Data) throws -> Void) throws -> Void) throws {
+    func append(data: Data) throws {
         let fileHandle = try FileHandle(forWritingTo: url)
-        defer { fileHandle.closeFile() }
 
-        try objcExceptionHandler.rethrowToSwift {
-            fileHandle.seekToEndOfFile()
-        }
+        if #available(iOS 13.0, *) {
+            try fileHandle.seekToEnd()
+            try fileHandle.write(contentsOf: data)
+            try fileHandle.close()
```

Comment:
i guess we still need to `defer { try file.close() }`, no?

Reply:
Hmm, inside `defer {}` we can't `try`. The trade-off will be `defer { try? file.close }`, but I think it's still better so will do it 👌.